### PR TITLE
fix(curriculum): tests incorrectly checking for `+ 1`

### LINKED ISSUE #55583: Role playing game - Step 124 - tests incorrectly checking for `+ 1` and hint has wrong variable name
### Describe the Issue

step 124 of the rpg game allows the code below to be submitted as the correct result even though it is missing the `+ 1` to make the random number start at 1. 

### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-basic-javascript-by-building-a-role-playing-game/step-124

### Your code

```
  monsterHealth -= weapons[currentWeaponIndex].power + Math.floor(Math.random() * xp);
```

### Expected behavior

test should look for the generated random number to start at 1

### Screenshots

_No response_

### System

N/A

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by dwrik**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #55583

#### Previous

fcc-original

#### Current

Fail scenarios
fcc-modified-fail

Pass scenarios
image
fcc-modified-pass2

### CODE REVIEW

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62a8e4dc6a60f85bf256a0cb.md
Comment (gikf) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62a8e4dc6a60f85bf256a0cb.md:57]:
```suggestion
assert.match(attack.toString(), /Math\.floor\(\s*Math\.random\(\s*\)\s*\*\s*xp\s*(\+\s*1\s*\)|\)\s*\+\s*1)/);
```
This can be a bit simpler.

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62a8e4dc6a60f85bf256a0cb.md
Comment (gikf) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62a8e4dc6a60f85bf256a0cb.md:60]:
```suggestion
You should add the result of `Math.floor(Math.random() * xp) + 1` to the result of `weapons[currentWeaponIndex].power`.
```
Hint update, as in https://github.com/freeCodeCamp/freeCodeCamp/issues/55583#issuecomment-2241605303

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62a8e4dc6a60f85bf256a0cb.md
Comment (gikf) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62a8e4dc6a60f85bf256a0cb.md:63]:
```suggestion
assert.match(attack.toString(), /monsterHealth\s*-=\s*weapons\s*\[\s*currentWeaponIndex\s*\]\s*\.power\s*\+\s*Math\.floor\(\s*Math\.random\(\s*\)\s*\*\s*xp\s*(\+\s*1\s*\)|\)\s*\+\s*1)/);
```

**Comment by dwrik** [comment]
@gikf

Thanks for the review. I'll go ahead and commit the suggested changes and resolve the comments.

**Comment by yuik371** [comment]
Was this helpful?

**Review by gikf — APPROVED** [review]

**Comment by dwrik** [comment]
@gikf

Thanks for the approval, but it still requires a second approval before it can be merged. Could someone else maybe take a look and provide the additional review needed?

**Review by a2937 — APPROVED** [review]
Looks Good To Me! 👍

**Comment by gikf** [comment]
Hi @dwrik. Congrats on your first pull request (PR)! 🎉

Thank you for your contribution to the page! 👍
We are happy to accept these changes and look forward to future contributions. 📝

**Comment by dwrik** [comment]
@gikf

Thank you! I'm glad to contribute. Looking forward to more opportunities to help out.

Best,
dwrik

**Pull request merged**
